lib/events.js

```diff
@@ -103,30 +103,29 @@ EventEmitter.prototype.getMaxListeners = function getMaxListeners() {
   return $getMaxListeners(this);
 };
 
-// Returns the longest sequence of `a` that fully appears in `b`,
-// of length at least 3.
-// This is a lazy approach but should work well enough, given that stack
-// frames are usually unequal or otherwise appear in groups, and that
-// we only run this code in case of an unhandled exception.
-function longestSeqContainedIn(a, b) {
-  for (var len = a.length; len >= 3; --len) {
-    for (var i = 0; i < a.length - len; ++i) {
-      // Attempt to find a[i:i+len] in b
-      for (var j = 0; j < b.length - len; ++j) {
-        let matches = true;
-        for (var k = 0; k < len; ++k) {
-          if (a[i + k] !== b[j + k]) {
-            matches = false;
-            break;
-          }
+// Returns the length and line number of the first sequence of `a` that fully
+// appears in `b` with a length of at least 4.
+function identicalSequenceRange(a, b) {
+  for (var i = 0; i < a.length - 3; i++) {
```

Comment:
I would cache `a.length - 3` in a variable.

Reply:
why?

Reply:
I expect the value to be constant fold (but I did not check).

Reply:
a could be quite big, and it used to be slightly faster to cache the value if it's computed.

Reply:
`a` should normally be small (we currently only use this for stack frames) and this implementation should also be faster than the one before. If it's about performance, I could save a couple comparisons by using a simple for loop instead of `indexOf` (currently I check until the last entry but the last three entries are not interesting).

@bmeurer do values like these get constant fold?